diff --git a/node_modules/@rbxts/planck-matter-hooks/out/hooks/log.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/hooks/log.d.ts
index 2c4f8d4..05437de 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/hooks/log.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/hooks/log.d.ts
@@ -1,2 +1 @@
-declare const log: (message: string) => void;
-export = log;
\ No newline at end of file
+export declare const log: (message: string) => void;
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useDeltaTime.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useDeltaTime.d.ts
index baf2574..554fba3 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useDeltaTime.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useDeltaTime.d.ts
@@ -1,4 +1,3 @@
 
 
-declare const useDeltaTime: () => number;
-export = useDeltaTime;
\ No newline at end of file
+export declare const useDeltaTime: () => number;
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useEvent.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useEvent.d.ts
index 1c8cd67..2b3734e 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useEvent.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useEvent.d.ts
@@ -1,3 +1,6 @@
+type PickEvent<T> = T extends RBXScriptSignal<infer U> ? U : never
+type Events<T extends Instance> = {
+    [key in InstanceEventNames<T>]: PickEvent<T[key]>;
+};
 
-declare const useEvent: <T extends unknown[]>(instance: Instance, event: string) => IterableFunction<T>;
-export = useEvent;
\ No newline at end of file
+export declare function useEvent<T extends Instance, E extends InstanceEventNames<T>>(instance: T, event: E): IterableFunction<LuaTuple<[number, ...Parameters<Events<T>[E]>]>>;
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useThrottle.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useThrottle.d.ts
index 8e87b6a..3010895 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/hooks/useThrottle.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/hooks/useThrottle.d.ts
@@ -1,3 +1,2 @@
 
-declare const useThrottle: (seconds: number, discriminator: unknown) => boolean;
-export = useThrottle;
\ No newline at end of file
+export declare const useThrottle: (seconds: number, discriminator: unknown) => boolean;
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/index.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/index.d.ts
index 4faf9f0..77c84e6 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/index.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/index.d.ts
@@ -1,7 +1,19 @@
-import { Plugin as PluginInterface, Scheduler } from "@rbxts/planck/out/types";
 
+import Scheduler from "@rbxts/planck/out/Scheduler";
+import { Plugin as Plug } from "@rbxts/planck/out/types";
+import { topoRuntime } from "./topoRuntime";
 
-export declare class Plugin implements PluginInterface {
-    constructor(module?: ModuleScript)
-    build<T extends unknown[]>(scheduler: Scheduler<T>): void;
+export declare const start: typeof topoRuntime.start
+export declare const useCurrentSystem: typeof topoRuntime.useCurrentSystem
+export declare const useFrameState: typeof topoRuntime.useFrameState
+export declare const useHookState: typeof topoRuntime.useHookState
+export declare const withinTopoContext: typeof topoRuntime.withinTopoContext
+
+export * from "./hooks/log"
+export * from "./hooks/useDeltaTime"
+export * from "./hooks/useEvent"
+export * from "./hooks/useThrottle"
+
+export declare class Plugin implements Plug {
+    build(schedular: Scheduler<unknown[]>): void
 }
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/init.lua b/node_modules/@rbxts/planck-matter-hooks/out/init.lua
index e269d7a..23314a8 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/init.lua
+++ b/node_modules/@rbxts/planck-matter-hooks/out/init.lua
@@ -1,5 +1,3 @@
-local ReplicatedStorage = game:GetService("ReplicatedStorage")
-
 type SystemInfo = {
 	name: string,
 	phase: any,
@@ -28,11 +26,6 @@ type SystemsReplace = {
 	old: SystemInfo,
 }
 
-type Plugin = {
-	build: (self: Plugin, scheduler: any) -> (),
-	new: (module: ModuleScript?) -> Plugin,
-}
-
 type ConnectionObject = {
 	Disconnect: (() -> ())?,
 	Destroy: (() -> ())?,
@@ -51,17 +44,6 @@ type CustomEvent = {
 	[any]: any,
 }
 
-type Library = {
-	useDeltaTime: () -> number,
-	useEvent: (
-		instance: Instance | { [string]: CustomEvent } | CustomEvent,
-		event: string | RBXScriptSignal | CustomEvent
-	) -> () -> (number, ...any),
-	useThrottle: (seconds: number, discriminator: any?) -> boolean,
-
-	Plugin: Plugin,
-}
-
 local topoRuntime
 local export = {} :: any
 
@@ -81,23 +63,16 @@ local function setHooks(package)
 	end
 end
 
--- When using Matter, we must use the topoRuntime in the library
-if ReplicatedStorage:FindFirstChild("Packages") then
-	for _, package in ReplicatedStorage.Packages._Index:GetChildren() do
-		if
-			string.find(package.Name, "matter%-ecs_matter")
-			or string.find(package.Name, "evaera_matter")
-		then
-			setHooks(package)
-			break
-		end
-	end
-end
-
 if not topoRuntime then
 	setHooks(script)
 end
 
+export.start = topoRuntime.start
+export.useCurrentSystem = topoRuntime.useCurrentSystem
+export.useFrameState = topoRuntime.useFrameState
+export.useHookState = topoRuntime.useHookState
+export.withinTopoContext = topoRuntime.withinTopoContext
+
 local Plugin = {}
 Plugin.__index = Plugin
 
@@ -112,7 +87,7 @@ function Plugin:build(scheduler: any)
 
 	local phaseDetails = {}
 
-	for _, phase in scheduler._orderedPhases do
+	for phase, _ in scheduler._phaseToSystems do
 		if not phaseDetails[phase] then
 			phaseDetails[phase] = {
 				lastTime = os.clock(),
@@ -250,4 +225,34 @@ end
 
 export.Plugin = Plugin
 
+type SchedulerLike<U...> = {
+	addPlugin: (
+		self: SchedulerLike<U...>,
+		plugin: Plugin<U...>
+	) -> SchedulerLike<U...>,
+	new: (U...) -> SchedulerLike<U...>,
+	[any]: any,
+}
+
+type Plugin<U...> = {
+	getLoop: (self: Plugin<U...>) -> {
+		[any]: any,
+	},
+	build: (self: Plugin<U...>, scheduler: SchedulerLike<U...>) -> (),
+	cleanup: ((self: Plugin<U...>) -> ())?,
+	new: (module: ModuleScript?) -> Plugin<U...>,
+}
+
+type Library = {
+	useDeltaTime: () -> number,
+	useEvent: (
+		instance: Instance | { [string]: CustomEvent } | CustomEvent,
+		event: string | RBXScriptSignal | CustomEvent
+	) -> () -> (number, ...any),
+	useThrottle: (seconds: number, discriminator: any?) -> boolean,
+	log: (...any) -> (),
+
+	Plugin: Plugin<...any>,
+}
+
 return export :: Library
diff --git a/node_modules/@rbxts/planck-matter-hooks/out/topoRuntime.d.ts b/node_modules/@rbxts/planck-matter-hooks/out/topoRuntime.d.ts
index ca94e37..18629fa 100644
--- a/node_modules/@rbxts/planck-matter-hooks/out/topoRuntime.d.ts
+++ b/node_modules/@rbxts/planck-matter-hooks/out/topoRuntime.d.ts
@@ -1,11 +1,9 @@
-
 interface TopoRuntime {
     start: (node: unknown, fn: (...args: unknown[]) => unknown) => void;
-    useHookState: <T extends unknown[]>(discriminator: unknown, cleanup: (...args: T) => void) => T;
+    useHookState: <T extends object>(discriminator: unknown, cleanup: (...args: unknown[]) => void) => T;
     useFrameState: () => unknown;
     useCurrentSystem: () => unknown;
     withinTopoContext: () => boolean;
 }
 
-declare const topoRuntime: TopoRuntime;
-export = topoRuntime;
\ No newline at end of file
+export declare const topoRuntime: TopoRuntime;
